src/store/audio.ts

```diff
@@ -333,57 +322,99 @@ export const audioStore = createPartialStore<AudioStoreTypes>({
-            });
-          const styles = getStyles(speaker, speakerInfo);
-          if (singer && singerInfo) {
-            styles.push(...getStyles(singer, singerInfo));
+            speakerStylePromise = speakerInfoPromise.then((speakerInfo) =>
+              getStyles(speaker, speakerInfo),
+            );
+          }
+
+          let singerInfoPromise: Promise<SpeakerInfo> | undefined = undefined;
+          let singerStylePromise: Promise<StyleInfo[]> | undefined = undefined;
+          if (singer != undefined) {
+            singerInfoPromise = instance
+              .invoke("singerInfoSingerInfoGet")({
+                speakerUuid: singer.speakerUuid,
+              })
+              .catch((error) => {
+                window.backend.logError(error, `Failed to get singerInfo.`);
+                throw error;
+              });
+            singerStylePromise = singerInfoPromise.then((singerInfo) =>
+              getStyles(singer, singerInfo),
+            );
           }
+
+          const baseSpeaker = speaker ?? singer;
+          if (baseSpeaker == undefined) {
+            throw new Error("assert baseSpeaker != undefined");
+          }
+          const baseCharacterInfo = await (speakerInfoPromise ??
+            singerInfoPromise);
+          if (baseCharacterInfo == undefined) {
+            throw new Error("assert baseSpeakerInfo != undefined");
+          }
+
+          const stylesPromise = Promise.all([
+            speakerStylePromise ?? [],
+            singerStylePromise ?? [],
+          ]).then((styles) => styles.flat());
+
           const characterInfo: CharacterInfo = {
-            portraitPath: base64ImageToUri(speakerInfo.portrait),
+            portraitPath: base64ImageToUri(baseCharacterInfo.portrait),
             metas: {
-              speakerUuid: SpeakerId(speaker.speakerUuid),
-              speakerName: speaker.name,
-              styles,
-              policy: speakerInfo.policy,
+              speakerUuid: SpeakerId(baseSpeaker.speakerUuid),
+              speakerName: baseSpeaker.name,
+              styles: await stylesPromise,
+              policy: baseCharacterInfo.policy,
             },
           };
           return characterInfo;
         };
-        const characterInfos: CharacterInfo[] = await Promise.all(
-          speakers.map(async (speaker) => {
-            return await getSpeakerInfo(speaker);
-          }),
+
+        const [speakers, singers] = await Promise.all([
+          instance.invoke("speakersSpeakersGet")({}),
+          state.engineManifests[engineId].supportedFeatures.sing
+            ? await instance.invoke("singersSingersGet")({})
+            : [],
+        ]).catch((error) => {
+          window.backend.logError(error, `Failed to get Speakers.`);
+          throw error;
+        });
+
+        const allUuids = new Set([
+          ...speakers.map((speaker) => speaker.speakerUuid),
+          ...singers.map((singer) => singer.speakerUuid),
+        ]);
```

Comment:
順番が変わらないことを前提にしてる、ということをコメントで追記してあげると将来のバグを防げるかもなので、お願いできると！
こんな感じ？
```suggestion
        // エンジン側の順番を保ってCharacterInfoを作る
        const allUuids = new Set([
          ...speakers.map((speaker) => speaker.speakerUuid),
          ...singers.map((singer) => singer.speakerUuid),
        ]);
```